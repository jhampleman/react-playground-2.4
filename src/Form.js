import React, { Component } from "react";
import Sessions from "./Sessions";
import { store } from './redux';

export default class Form extends Component {
  constructor(props) {
    super(props);

    this.state = {
      myFirstName: "",
      myLastName: ""
    };
  }

  submitForm = e => {
    e.preventDefault();

    this.props.handleData(this.state);
        console.log(store.getState());
  
  };
  
   clearCache = e => {
    e.preventDefault();
    //CLEAR local storage (cache) - uncomment line below.
    localStorage.clear();
  };


  handleNameChange = e => {
    //console.log(e.target.value);
    this.setState({
      myFirstName: e.target.value
    });
  };

  handleEmailChange = e => {
    this.setState({
      myLastName: e.target.value
    });
  };

  render() {
    return (
      <div>
        <p className="blue enztest">This is in Form.js and uses Enzyme - See Form.spec</p>
        <form>
          First Name:{" "}
          <input
            type="text"
            name="myFirstName"
            onChange={this.handleNameChange}
          />
          <br />
          Last Name:{" "}
          <input
            type="text"
            name="myLastName"
            onChange={this.handleEmailChange}
          />
          <br />
          <br />
          <input type="button" value="Submit" onClick={this.submitForm} />
          
          <Sessions />
          <input type="button" value="Clear Cache" onClick={this.clearCache} />
          
        </form>
      </div>
    );
  }
}
