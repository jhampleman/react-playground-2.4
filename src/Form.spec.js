import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form', () => {
  let wrapper;
  beforeEach(() => wrapper = shallow(<Form />));
it('should render correctly', () => expect(wrapper).toMatchSnapshot());
  it('should render a <div />', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });
  it('should render the Form component', () => {
  expect(wrapper.containsAllMatchingElements([
        <p className="blue enztest">This is in Form.js and uses Enzyme - See Form.spec</p>
  ])).toEqual(true);
});
});