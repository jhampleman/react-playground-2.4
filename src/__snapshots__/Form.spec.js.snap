// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="blue enztest"
        >
          This is in Form.js and uses Enzyme - See Form.spec
        </p>,
        <form>
          First Name:
           
          <input
            name="myFirstName"
            onChange={[Function]}
            type="text"
          />
          <br />
          Last Name:
           
          <input
            name="myLastName"
            onChange={[Function]}
            type="text"
          />
          <br />
          <br />
          <input
            onClick={[Function]}
            type="button"
            value="Submit"
          />
          <Sessions />
          <input
            onClick={[Function]}
            type="button"
            value="Clear Cache"
          />
        </form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "This is in Form.js and uses Enzyme - See Form.spec",
          "className": "blue enztest",
        },
        "ref": null,
        "rendered": "This is in Form.js and uses Enzyme - See Form.spec",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "First Name:",
            " ",
            <input
              name="myFirstName"
              onChange={[Function]}
              type="text"
            />,
            <br />,
            "Last Name:",
            " ",
            <input
              name="myLastName"
              onChange={[Function]}
              type="text"
            />,
            <br />,
            <br />,
            <input
              onClick={[Function]}
              type="button"
              value="Submit"
            />,
            <Sessions />,
            <input
              onClick={[Function]}
              type="button"
              value="Clear Cache"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          "First Name:",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "myFirstName",
              "onChange": [Function],
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          "Last Name:",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "myLastName",
              "onChange": [Function],
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onClick": [Function],
              "type": "button",
              "value": "Submit",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onClick": [Function],
              "type": "button",
              "value": "Clear Cache",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="blue enztest"
          >
            This is in Form.js and uses Enzyme - See Form.spec
          </p>,
          <form>
            First Name:
             
            <input
              name="myFirstName"
              onChange={[Function]}
              type="text"
            />
            <br />
            Last Name:
             
            <input
              name="myLastName"
              onChange={[Function]}
              type="text"
            />
            <br />
            <br />
            <input
              onClick={[Function]}
              type="button"
              value="Submit"
            />
            <Sessions />
            <input
              onClick={[Function]}
              type="button"
              value="Clear Cache"
            />
          </form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "This is in Form.js and uses Enzyme - See Form.spec",
            "className": "blue enztest",
          },
          "ref": null,
          "rendered": "This is in Form.js and uses Enzyme - See Form.spec",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "First Name:",
              " ",
              <input
                name="myFirstName"
                onChange={[Function]}
                type="text"
              />,
              <br />,
              "Last Name:",
              " ",
              <input
                name="myLastName"
                onChange={[Function]}
                type="text"
              />,
              <br />,
              <br />,
              <input
                onClick={[Function]}
                type="button"
                value="Submit"
              />,
              <Sessions />,
              <input
                onClick={[Function]}
                type="button"
                value="Clear Cache"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "First Name:",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "myFirstName",
                "onChange": [Function],
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "Last Name:",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "myLastName",
                "onChange": [Function],
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onClick": [Function],
                "type": "button",
                "value": "Submit",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "onClick": [Function],
                "type": "button",
                "value": "Clear Cache",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
